### PR TITLE
Move hapi to python/paddle root dir.

### PR DESCRIPTION
**Pull request by qingqing01**
### PR types

Function optimization

### PR changes

APIs

### Describe

- Move hapi to python/paddle root dir.
- Refine some code.

1.  tree dir:

```
paddle/
|-- hapi
|   `-- model.py  callbacks.py  dygraph_layer_patch.py progressbar.py logger.py
|-- utils
|--`--download.py
|--text
|--`--text.py
`-- vision
    |-- datasets
    |-- models
    `-- transforms
```

2.  Move DistributedBatchSampler to paddle/fluid/dataloader/batch_sampler.py (`paddle.io.DistributedBatchSampler`)

3. Remove 5 files

python/paddle/utils/plotcurve.py
python/paddle/utils/preprocess_img.py
python/paddle/utils/preprocess_util.py
python/paddle/utils/show_pb.py
python/paddle/utils/torch2paddle.py 

4. After remove `hapi`, some usage:

```
import paddle

paddle.Model
import paddle.satic.InputSpec  as InputSpec
paddle.metric.Accuracy
paddle.vision.datasets.MNIST       # paddle.vision.MNIST
paddle.vision.transforms.Resize  # paddle.vision.Resize
paddle.vision.models.ResNet       # paddle.vision.ResNet
paddle.text.datasets.Imdb            # paddle.text.Imdb
```

### CODE REVIEW

**Review by LielinJiang — DISMISSED** [review]

**Review by heavengate — DISMISSED** [review]

**Review by guoshengCS — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/vision/datasets/cifar.py [hunk 1]
```diff
@@ -58,33 +58,33 @@ class Cifar10(Dataset):
 
         .. code-block:: python
 
-	    import paddle
-	    from paddle.incubate.hapi.datasets import Cifar10
-	    from paddle.incubate.hapi.vision.transforms import Normalize
+            import paddle
+            from paddle.incubate.hapi.datasets import Cifar10
+            from paddle.incubate.hapi.vision.transforms import Normalize
```
Comment (saxon-zh) [inline python/paddle/vision/datasets/cifar.py]: 这两个目录没有变

File: python/paddle/vision/datasets/cifar.py [hunk 2]
```diff
@@ -164,33 +164,33 @@ class Cifar100(Cifar10):
 
         .. code-block:: python
 
-	    import paddle
-	    from paddle.incubate.hapi.datasets import Cifar100
-	    from paddle.incubate.hapi.vision.transforms import Normalize
+            import paddle
+            from paddle.incubate.hapi.datasets import Cifar100
+            from paddle.incubate.hapi.vision.transforms import Normalize
```
Comment (saxon-zh) [inline python/paddle/vision/datasets/cifar.py]: 目录未调整

File: python/paddle/vision/datasets/voc2012.py [hunk 3]
```diff
@@ -52,28 +52,28 @@ class VOC2012(Dataset):
 
         .. code-block:: python
 
-	    import paddle
-	    from paddle.incubate.hapi.datasets import VOC2012
+            import paddle
+            from paddle.incubate.hapi.datasets import VOC2012
```
Comment (saxon-zh) [inline python/paddle/vision/datasets/voc2012.py]: 目录未调整

File: python/paddle/hapi/callbacks.py [hunk 4]
```diff
@@ -292,23 +294,22 @@ class ProgBarLogger(Callback):
...
-            inputs = [hapi.Input([-1, 1, 28, 28], 'float32', 'image')]
-            labels = [hapi.Input([None, 1], 'int64', 'label')]
+            inputs = [InputSpec([-1, 1, 28, 28], 'float32', 'image')]
+            labels = [InputSpec([None, 1], 'int64', 'label')]
 
-            train_dataset = hapi.datasets.MNIST(mode='train')
+            train_dataset = paddle.vision.datasets.MNIST(mode='train')
 
-            model = hapi.Model(hapi.vision.LeNet(classifier_activation=None),
+            model = paddle.Model(paddle.vision.LeNet(classifier_activation=None),
```
Comment (saxon-zh) [inline python/paddle/hapi/callbacks.py:304]: 路径错误paddle.vision.models.

File: python/paddle/hapi/callbacks.py [hunk 5]
```diff
@@ -428,23 +429,22 @@ class ModelCheckpoint(Callback):
...
```
Comment (saxon-zh) [inline python/paddle/hapi/callbacks.py:439]: 路径错误,paddle.vision.models

File: python/paddle/metric/metrics.py [hunk 6]
```diff
@@ -202,12 +202,11 @@ class Accuracy(Metric):
...
 
         import paddle
-        import paddle.incubate.hapi as hapi
 
         paddle.disable_static()
-        train_dataset = hapi.datasets.MNIST(mode='train')
+        train_dataset = paddle.vision.datasets.MNIST(mode='train')
 
-        model = hapi.Model(hapi.vision.LeNet(classifier_activation=None))
+        model = paddle.Model(paddle.vision.LeNet(classifier_activation=None))
```
Comment (saxon-zh) [inline python/paddle/metric/metrics.py:209]: 路径错误，paddle.vision.models.

File: python/paddle/vision/datasets/cifar.py (hunk 2 above)
Comment (qingqing01) [reply python/paddle/vision/datasets/cifar.py]: Done. Thanks!

File: python/paddle/vision/datasets/voc2012.py (hunk 3 above)
Comment (qingqing01) [reply python/paddle/vision/datasets/voc2012.py]: Done. Thanks!

File: python/paddle/hapi/callbacks.py (hunk 5 above)
Comment (qingqing01) [reply python/paddle/hapi/callbacks.py:439]: Done. Thanks!  另外， paddle.vision.LeNet用法 和 paddle.vision.models.LeNet用法都可以。

**Comment by qingqing01** [comment]
```
2020-08-29 23:44:41 current pr 26442 got approvals: FALSE
2020-08-29 23:44:41 ************************************
2020-08-29 23:44:41 It is forbidden to disable or delete the unit-test.
2020-08-29 23:44:41 
2020-08-29 23:44:41 If you must delete it temporarily, please add it to[https://github.com/PaddlePaddle/Paddle/wiki/Temporarily-disabled-Unit-Test].
2020-08-29 23:44:41 Then you must have one RD (kolinwei(recommended) or zhouwei25) approval for the deletion of unit-test. 
2020-08-29 23:44:41 
2020-08-29 23:44:41 If you have any problems about deleting unit-test, please read the specification [https://github.com/PaddlePaddle/Paddle/wiki/Deleting-unit-test-is-forbidden]. 
2020-08-29 23:44:41 
2020-08-29 23:44:41 Following unit-tests are deleted in this PR: 
2020-08-29 23:44:41  -  test_uncombined_weight2state_dict 
```

@kolinwei @zhouwei25 please help to review.  delete uncombined_weight2state_dict, so delete  uniting test.

**Comment by qingqing01** [comment]
```
2020-08-29 22:39:53 + assert_api_spec_approvals
2020-08-29 22:39:53 + /bin/bash /workspace/Paddle/tools/check_api_approvals.sh
2020-08-29 22:39:53   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2020-08-29 22:39:53                                  Dload  Upload   Total   Spent    Left  Speed
2020-08-29 22:39:53 
2020-08-29 22:39:54   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2020-08-29 22:39:55   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2020-08-29 22:39:56   0 20077    0    70    0     0     36      0  0:09:17  0:00:01  0:09:16    36
2020-08-29 22:39:56 100 20077  100 20077    0     0   8995      0  0:00:02  0:00:02 --:--:--  8999
2020-08-29 22:40:08 ****************
2020-08-29 22:40:08 0. You must have one RD (XiaoguangHu01 or lanxianghit) and one TPM (saxon-zh or jzhang533 or swtkiwi or Heeenrrry or TCChenlong) approval for the api change for the management reason of API interface.
2020-08-29 22:40:08 1. You must have one TPM (saxon-zh or jzhang533 or swtkiwi or Heeenrrry or TCChenlong) approval for the api change for the management reason of API document.
2020-08-29 22:40:08 2. You must have one RD (zhiqiu (Recommend) or phlrain) approval for the api change for the opreator-related api without 'core.ops'.
```

@XiaoguangHu01  @saxon-zh @jzhang533 @zhiqiu @phlrain  please help to review.

**Comment by qingqing01** [comment]
```
2020-08-29 21:53:24 ****************
2020-08-29 21:53:24 0. You must have Dianhai approval for change 20+ files or add than 1000+ lines of content.
2020-08-29 21:53:24 1. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE to EXCLUSIVE or DIST, or setting TIMEOUT properties.
2020-08-29 21:53:24 The corresponding lines are as follows:
2020-08-29 21:53:24 python/paddle/tests/CMakeLists.txt
2020-08-29 21:53:24 +      set_tests_properties(${TARGET_NAME} PROPERTIES TIMEOUT 600 LABELS "RUN_TYPE=DIST")
2020-08-29 21:53:24 For more information, please refer to:https://github.com/PaddlePaddle/Paddle/wiki/PaddlePaddle-Unit-test-specification
```

@raindrops2sea @XieYunshen  please help to review.

**Review by XieYunshen — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by zhwesky2010 — APPROVED** [review]
just remove useless unittest

**Review by kolinwei — APPROVED** [review]

**Review by saxon-zh — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
